src/Hl7.FhirPath/FhirPath/Functions/CollectionOperators.cs

```diff
@@ -87,5 +87,22 @@ public static IEnumerable<ITypedElement> Navigate(this ITypedElement element, st
                 return element.Children(name);
             }
         }
+
+        public static string FpJoin(this IEnumerable<ITypedElement> collection, string separator)
+        {
+            //if the collection is empty return the empty result
+            if (!collection.Any())
+                return string.Empty;
+
+            //only join collections with string values inside
+            foreach (var typedElement in collection)
+            {
+                if (typedElement.InstanceType != "string")
```

Comment:
@marcovisserFurore : Hi marco. At this place I didnt found any good solution to check this type better than that. In Primitive.cs the types get set hard coded as string. Maybe there is some enum or somethig that I didn´t see?